src/parser.ml

```diff
@@ -730,6 +818,33 @@ let is_empty st =
     set_pos st off;
     true
 
+let inline_attribute_string s =
+  let a =
+    match peek s with
+    | Some '{' ->
+        let buf = Buffer.create 64 in
+        let rec loop s pos =
+          match peek s with
+          | Some '}' ->
+              junk s;
+              Some (Buffer.contents buf)
+          | None | Some '{' ->
+              set_pos s pos; None
+          | Some c ->
+              Buffer.add_char buf c;
+              junk s;
+              loop s pos
+        in
+        junk s;
+        loop s (pos s)
+    | _ ->
+        None
+  in
+  try
+    parse_attributes a
+  with
+    | Invalid_argument _ -> empty_attributes
```

Comment:
Style: typically we put the exception pattern in the same line as `with` and the exception handler in the next different line.

Reply:
Also, which function raises `Invalid_argument` ?

Reply:
The call to s.[0] can throw it if the string is empty. Actually this is bad, it could make fail attributes which contains too much spaces, I'll just ignore empty strings

Reply:
OK, in general one should never catch the exception `Invalid_argument` which indicates programmer error; rather one should check explicitly for the condition that raises it (in this case, an empty string, etc).